File: old/basic/src/main/java/com/dinesh/android/kotlin/retrofit/my_first_api/MainActivity.java
package com.dinesh.android.kotlin.retrofit.my_first_api;

public class MainActivity {
}

/*


public class MainActivity extends AppCompatActivity {
    private static final String TAG = "log_test";

    ApiInterface apiInterface;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        apiInterface = ApiClient.getRetrofit().create(ApiInterface.class);

        postModel();
        postModelList();
        postByField();
    }

    private void postByField() {
        Call<Model> call = apiInterface.postByField("Dinesh","K","dk@gmail.com","8667024800","4");
        call.enqueue(new Callback<Model>() {
            @Override
            public void onResponse(Call<Model> call, Response<Model> response) {

                Log.i(TAG, "postModelList: " + response.isSuccessful());
                Log.i(TAG, "postModelList: " + response.code());
            }

            @Override
            public void onFailure(Call<Model> call, Throwable t) {
                Log.i(TAG, "postModelList onFailure: " + t.getLocalizedMessage());
            }
        });
    }


    private void postModelList() {
        Model model = new Model("Dinesh","K","dk@gmail.com","8667024800","4");
        Call<List<Model>> call = apiInterface.postModelList(model);
        call.enqueue(new Callback<List<Model>>() {
            @Override
            public void onResponse(Call<List<Model>> call, Response<List<Model>> response) {
                Log.i(TAG, "postModelList: " + response.isSuccessful());
                Log.i(TAG, "postModelList: " + response.code());
            }
            @Override
            public void onFailure(Call<List<Model>> call, Throwable t) {
                Log.i(TAG, "postModelList onFailure: " + t.getLocalizedMessage());
            }
        });
    }

    private void postModel() {
        Model model = new Model("Dinesh","K","dk@gmail.com","8667024800","4");
        Call<Model> call = apiInterface.postModel(model);
        call.enqueue(new Callback<Model>() {
            @Override
            public void onResponse(Call<Model> call, Response<Model> response) {
                Log.i(TAG, "onResponse: " + response.isSuccessful());
            }
            @Override
            public void onFailure(Call<Model> call, Throwable t) {
                Log.i(TAG, "postModel onFailure: " + t.getLocalizedMessage());
            }
        });
    }

}


 */

/*

public interface ApiInterface {
//    URL =192.168.35.52/testapi/register.php

    @POST("register.php")
    Call<Model> postModel(@Body Model todo);

    @POST("register.php")
    Call<List<Model>> postModelList(@Body Model todo);


    @POST("register.php")
    @FormUrlEncoded
    Call<Model> postByField(@Field("firstname") String firstname,
                        @Field("lastname") String lastname,
                        @Field("email") String email,
                        @Field("mobile") String mobile,
                        @Field("dob") String dob);


}

 */

/*

public class ApiClient {
//    192.168.35.52/testapi/register.php
    private static final String BASE_URL = "http://192.168.35.52/testapi/";
    private static Retrofit retrofit = null;

    public static Retrofit getRetrofit() {
        Gson gson = new GsonBuilder()
                .setLenient()
                .create();

        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient okHttpClient = new OkHttpClient.Builder()
                .addInterceptor(loggingInterceptor)
                .build();

        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
//                    .client(okHttpClient)
                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .build();
        }
        return retrofit;
    }
}

 */

/*

public class Model {

    @SerializedName("firstname")
    @Expose
    public String firstname;
    @SerializedName("lastname")
    @Expose
    public String lastname;
    @SerializedName("email")
    @Expose
    public String email;
    @SerializedName("mobile")
    @Expose
    public String mobile;
    @SerializedName("dob")
    @Expose
    public String dob;

    public Model(String firstname, String lastname, String email, String mobile, String dob) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.email = email;
        this.mobile = mobile;
        this.dob = dob;
    }

    public Model() {
    }

    @Override
    public String toString() {
        return "Model{" +
                "firstname='" + firstname + '\'' +
                ", lastname='" + lastname + '\'' +
                ", email='" + email + '\'' +
                ", mobile='" + mobile + '\'' +
                ", dob='" + dob + '\'' +
                '}';
    }

}

 */


/*

{
	"info": {
		"_postman_id": "413d0015-bd7e-425d-9c44-043af39b043b",
		"name": "testapi",
		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
		"_exporter_id": "8132379"
	},
	"item": [
		{
			"name": "192.168.35.52/testapi/register.php",
			"request": {
				"method": "POST",
				"header": [],
				"body": {
					"mode": "formdata",
					"formdata": [
						{
							"key": "firstname",
							"value": "testapi1",
							"type": "text"
						},
						{
							"key": "lastname",
							"value": "arjun",
							"type": "text"
						},
						{
							"key": "email",
							"value": "testapi@gmail.com",
							"type": "text"
						},
						{
							"key": "mobile",
							"value": "98745622",
							"type": "text"
						},
						{
							"key": "dob",
							"value": "1",
							"type": "text"
						}
					]
				},
				"url": {
					"raw": "192.168.35.52/testapi/register.php",
					"host": [
						"192",
						"168",
						"35",
						"52"
					],
					"path": [
						"testapi",
						"register.php"
					]
				}
			},
			"response": []
		}
	]
}

 */

File: old/basic/src/main/java/com/dinesh/android/kotlin/retrofit/authorization/bearer/MainActivity.kt
package com.dinesh.android.kotlin.retrofit.authorization.bearer

import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.result.PickVisualMediaRequest
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.dinesh.android.kotlin.retrofit.ApiClient
import kotlinx.coroutines.launch
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.MultipartBody
import okhttp3.RequestBody
import okhttp3.RequestBody.Companion.toRequestBody


private val TAG = "log_" + MainActivity::class.java.name.split(MainActivity::class.java.name.split(".").toTypedArray()[2] + ".").toTypedArray()[1]

class MainActivity : AppCompatActivity() {
    private lateinit var apiService: ApiService

    private val pickImage = registerForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri: Uri? ->
        if (uri != null) {
            uploadImage(uri)
        } else {
            Log.e(TAG, "Image URI is null")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        pickImage.launch(PickVisualMediaRequest())
    }

    private fun uploadImage(imageFileUri: Uri) {
        apiService = ApiClient.getApiInterface("https://chennainextlevel.com/")

        val imagePart = MultipartBody.Part.createFormData(
            "order_details[0][image]",
            "image.jpg",
            contentResolver.openInputStream(imageFileUri)!!.readBytes().toRequestBody("image/*".toMediaTypeOrNull())
        )

        val customerId = "38"
        val token = "81|rvf3ZWny9w88tXYAOWCwXh6VBNbTO5ZBHroSbW30"

        val orderName = "testing"
        val orderSpecification = "testing"
        val orderPrice = "150"
        val orderQuantity = "12"

        val acceptHeader = "application/json"
        val authorizationHeader = "Bearer $token"
        val customerIdPart = createPartFromString(customerId)
        val orderNamePart = createPartFromString(orderName)
        val orderSpecificationPart = createPartFromString(orderSpecification)
        val orderPricePart = createPartFromString(orderPrice)
        val orderQuantityPart = createPartFromString(orderQuantity)
        val orderImageUrlPart = createPartFromString("")

        val requestMessage = "Request: customerId=$customerId, orderName=$orderName, orderSpecification=$orderSpecification, " +
                "orderPrice=$orderPrice, orderQuantity=$orderQuantity"

        Log.d(TAG, requestMessage)
        Log.d(TAG, "Sending create order request...")
        lifecycleScope.launch {
            try {
                val response = apiService.createOrder(
                    acceptHeader,
                    authorizationHeader,
                    customerIdPart,
                    orderNamePart,
                    orderSpecificationPart,
                    orderPricePart,
                    orderQuantityPart,
                    orderImageUrlPart,
                    imagePart
                )

                if (response.isSuccessful) {
                    Log.d(TAG, "response ---> $response")
                    Log.d(TAG, "body ---> ${response.body()}")
                    Toast.makeText(this@MainActivity, response.body()?.message, Toast.LENGTH_LONG).show()
                } else {
                    Log.e(TAG, "Failed to create order: ${response.code()}")
                }
            } catch (e: Exception) {
                Log.e(TAG, "Network request failed: ${e.message}")
            }
        }
    }

    private fun createPartFromString(string: String): RequestBody {
        return string.toRequestBody("text/plain".toMediaTypeOrNull())
    }
}

//  pc_bearer_post.json
/*


{
	"info": {
		"_postman_id": "a3f04115-adb6-4c74-9946-c6c44683c9a5",
		"name": "onepixcel",
		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
		"_exporter_id": "27788171"
	},
	"item": [
		{
			"name": "Create Orders",
			"request": {
				"auth": {
					"type": "bearer",
					"bearer": [
						{
							"key": "token",
							"value": "81|rvf3ZWny9w88tXYAOWCwXh6VBNbTO5ZBHroSbW30",
							"type": "string"
						}
					]
				},
				"method": "POST",
				"header": [
					{
						"key": "Accept",
						"value": "application/json",
						"type": "text"
					}
				],
				"body": {
					"mode": "formdata",
					"formdata": [
						{
							"key": "customer_id",
							"value": "38",
							"type": "text"
						},
						{
							"key": "order_details[0][name]",
							"value": "testing",
							"type": "text"
						},
						{
							"key": "order_details[0][specification]",
							"value": "testing",
							"type": "text"
						},
						{
							"key": "order_details[0][price]",
							"value": "150",
							"type": "text"
						},
						{
							"key": "order_details[0][image]",
							"type": "file",
							"src": "/Users/gulshan/Downloads/home.png"
						},
						{
							"key": "order_details[1][name]",
							"value": "testing1",
							"type": "text"
						},
						{
							"key": "order_details[1][specification]",
							"value": "testing1",
							"type": "text"
						},
						{
							"key": "order_details[1][price]",
							"value": "250",
							"type": "text"
						},
						{
							"key": "order_details[1][image]",
							"type": "file",
							"src": "/Users/gulshan/Downloads/peakpx.jpg"
						},
						{
							"key": "order_details[0][quantity]",
							"value": "12",
							"type": "text"
						},
						{
							"key": "order_details[1][quantity]",
							"value": "15",
							"type": "text"
						},
						{
							"key": "order_details[0][imageurl]",
							"value": "",
							"type": "text"
						},
						{
							"key": "order_details[1][imageurl]",
							"value": "",
							"type": "text"
						}
					]
				},
				"url": {
					"raw": "https://chennainextlevel.com/api/v1/orders",
					"protocol": "https",
					"host": [
						"chennainextlevel",
						"com"
					],
					"path": [
						"api",
						"v1",
						"orders"
					]
				}
			},
			"response": []
		}
	]
}


 */